va/win32/va_win32.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Microsoft Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "sysdeps.h"
+#include "va.h"
+#include "va_backend.h"
+#include "va_internal.h"
+#include "va_trace.h"
+#include "va_win32.h"
+
+static int va_DisplayContextIsValid(
+    VADisplayContextP pDisplayContext
+)
+{
+    return (pDisplayContext != NULL &&
+            pDisplayContext->pDriverContext != NULL);
+}
+
+static void va_DisplayContextDestroy(
+    VADisplayContextP pDisplayContext
+)
+{
+    if (pDisplayContext == NULL)
+        return;
+
+    if (pDisplayContext->pDriverContext
+        && pDisplayContext->pDriverContext->native_dpy)
+        free(pDisplayContext->pDriverContext->native_dpy);
+
+    if (pDisplayContext->pDriverContext)
+        free(pDisplayContext->pDriverContext);
+
+    free(pDisplayContext);
+}
+
+static VAStatus va_DisplayContextGetNumCandidates(
+    VADisplayContextP pDisplayContext,
+    int *num_candidates
+)
+{
+    LUID* adapter = pDisplayContext->pDriverContext->native_dpy;
+    if(adapter) 
+    {
+        // TODO: If adapter LUID is specified, query from
+        // some registry that maps drivers to adapter LUIDs
+        // or fallback to default driver otherwise
+        *num_candidates = 1;
```

Comment:
You can take a look at https://github.com/KhronosGroup/OpenCL-ICD-Loader/blob/main/loader/windows/icd_windows_dxgk.c for an example of how to look up drivers from the registry. You probably want to cache this during display context creation.